storage/client/client.go

```diff
@@ -1294,6 +1297,21 @@ func (c *StorageClient) RuntimeEvents(ctx context.Context, p apiTypes.GetRuntime
 		h := ethCommon.HexToHash(*p.EvmLogSignature)
 		evmLogSignature = &h
 	}
+	if p.NftId != nil && p.ContractAddress == nil {
+		return nil, fmt.Errorf("must specify contract_address with nft_id")
+	}
```

Comment:
Can we add a check to make sure that `type`/`evm_log_signature` weren't specified to be anything other than `evm.log`/`erc721Transfer_signature` in the query if `NftId` was provided? Otherwise such a query would return no results because the filters clash, but it wouldn't be obvious to the user why this happened.

```
	if p.NftId != nil && (p.Type != 'evm.log' || evmLogSignature != 'erc 721 signature here') {
	        return nil, fmt.Errorf("nft_id field currently only evm erc-721 transfer events")
	 }
```

Reply:
Because they would want to list e.g. the `Allow` calls for a certain `NftId`.
Or, here's a related not-great scenario: User will not specify any `evm_log_signature` (which seems to be the expected usage right now) and then will be surprised that they only get the Transfer events, not also Allow.

In this [other thread](https://github.com/oasisprotocol/nexus/pull/579#discussion_r1418351008), what I thought I was agreeing to was to "[...] support arbitrary event signatures", which in my understanding was to remove this from SQL: `evs.evm_log_signature = '\xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'`.

I still think we should either:
- Remove that thing from SQL (but that might be prohibitively slow, and also semantics are sketchy because we can't be sure that `(body -> 'topics' ->> 3)` is the ntf id), OR
- Require the query to also specify `evm_log_signature`, as Andy is suggesting. We can say "unsupported value" to any value other than `erc721Transfer_signature` for now, and later expand to allowing other values.

If the above+Andy convinces you, great. If not, I've lost the wind in my sails to belabor this relatively minor point further, convinced as I might be. Therefore LGTMing.

Reply:
let's try adding support for Allow events. you meant events, not calls right? this wasn't meant to limit the API to transfers. we just hadn't implemented it

Reply:
ugh though wasn't this nft_id field documented as something specifically for ERC-721 transfer events? that's going to be on them for filtering for ERC-721 transfer events that aren't ERC-721 transfer events

Reply:
omg that's been removed since the comment on it was posted. did anyone ask for that? should I put it back :skull:

Reply:
ok in conclusion:

- don't wanna
- adding a note to the description though